Misscalculated Debye-Waller factor in ComputeIncoherentDos.py

<!-- Issues which impact security should have a Vulnerability label added to them -->

<!-- If this issue is raised on behalf of a user they should be named here.
**Original reporter:** [username facility]/[nobody]
-->

**Describe the bug**
I noticed that the Debye-Waller factor in the algorithm ComputeIncoherentDos.py is calculated as:
` DWF = np.exp(-2*(qqgridsq*msd))` with `msd` the averaged mean square displacement provided by the user. 

Given the definition from the in the reference page (https://docs.mantidproject.org/nightly/algorithms/ComputeIncoherentDOS-v1.html#algm-computeincoherentdos), the DWF should be:
` DWF = np.exp(-(qqgridsq*msd))`

I think that the factor 2 should not be here.

> @jhaigh0 @fucaline
> 
> Yes you're right that the factor of 2 should not be there in the code.
> 
> Sorry for the long delay getting back to you - I had a bit of a detour looking into different notations in use.
> 
> @jhaigh0 when you fix this could you also amend the documentation to say that the `msd` parameter ![\langle\hat{u}^2\rangle](https://camo.githubusercontent.com/3e798ab007985823da4a57ec790f3e4f5f8dc163fc30c6cd78f20cee5360ed9e/68747470733a2f2f6c617465782e636f6465636f67732e636f6d2f7376672e696d6167653f2535436c616e676c65253543686174253742752537442535453225354372616e676c65) is also called the isotropic atomic displacement parameter `Uiso` ![U_{\mathrm{iso}}](https://camo.githubusercontent.com/3d97edbce0cc0348e32462eb057919bb910e9e7558c39ae8c3eea7d2da6e891a/68747470733a2f2f6c617465782e636f6465636f67732e636f6d2f7376672e696d6167653f555f2537422535436d617468726d25374269736f253744253744) in Rietveld refinement programs like [GSAS](https://subversion.xray.aps.anl.gov/trac/EXPGUI/wiki/GSASIntro).
> 
> Could you also say in the docs that there is also another type of atomic displacement parameters used in programs like [FullProf](https://www.ill.eu/sites/fullprof/) called `Biso` which is related by:
> 
> ![B_{\mathrm{iso}}=8\pi^2U_{\mathrm{iso}}](https://camo.githubusercontent.com/5d8c84f3333f79fab513d7f2fb89fade6b49881446b36ea9faf8c40354d14e49/68747470733a2f2f6c617465782e636f6465636f67732e636f6d2f7376672e696d6167653f425f2537422535436d617468726d25374269736f2537442537443d38253543706925354532555f2537422535436d617468726d25374269736f253744253744)
> 
> The source for this information is page 123 of the [GSAS manual](https://subversion.xray.aps.anl.gov/EXPGUI/gsas/all/GSAS%20Manual.pdf).
> 
> My confusion is because whilst these are the parameters which are actually used by crystallography programs, the International Tables for Crystallography actually defines a different parameter given by:
> 
> ![U_{\mathrm{eqiv}}=\frac{1}{3}\langle\hat{u}^2\rangle](https://camo.githubusercontent.com/8d33a4f80c40f97a0e7878ba5dbc20e8323fe029731b0c0dd0768c8cd1d0ccd1/68747470733a2f2f6c617465782e636f6465636f67732e636f6d2f7376672e696d6167653f555f2537422535436d617468726d253742657169762537442537443d2535436672616325374231253744253742332537442535436c616e676c65253543686174253742752537442535453225354372616e676c65)
> 
> As defined by equation [52] in [this paper](https://www.iucr.org/resources/commissions/crystallographic-nomenclature/adp).

Hi @mducle thanks for the reply, on removing the factor of two `ComputeIncoherentDOSTest::test_nonzero_MDS` fails like this:
```
Traceback (most recent call last):
  File "C:\Users\bya67386\work\mantid\Framework\PythonInterface\test\python\plugins\algorithms\ComputeIncoherentDOSTest.py", line 226, in test_nonzero_MDS
    self.assertAlmostEqual(dos_Ys[i], g[i], delta=g[i] * 1e-12)
AssertionError: 1743788954116.2805 != 7.538620364050139e+24 within 7538620364050.139 delta (7.538620364048395e+24 difference)
```

I think that [this line](https://github.com/mantidproject/mantid/blob/main/Framework/PythonInterface/test/python/plugins/algorithms/ComputeIncoherentDOSTest.py#L49) will have to change to fix this. Could you give me a pointer as to how it should be changed.

--- a/Framework/PythonInterface/plugins/algorithms/ComputeIncoherentDOS.py
+++ b/Framework/PythonInterface/plugins/algorithms/ComputeIncoherentDOS.py
@@ -271,7 +271,7 @@ class ComputeIncoherentDOS(PythonAlgorithm):
         engrid = np.transpose(np.tile(en, (np.shape(y)[1], 1)))
         # Calculates the Debye-Waller and Bose factors from the Temperature and mean-squared displacements
         qqgridsq = qqgrid**2
-        DWF = np.exp(-2 * (qqgridsq * msd))
+        DWF = np.exp(-(qqgridsq * msd))
         idm = np.where(engrid < 0)
         idp = np.where(engrid >= 0)
         # The expression for the population (Bose) factor and phonon energy dependence below actually refer to the phonon

--- a/Framework/PythonInterface/test/python/plugins/algorithms/ComputeIncoherentDOSTest.py
+++ b/Framework/PythonInterface/test/python/plugins/algorithms/ComputeIncoherentDOSTest.py
@@ -46,7 +46,7 @@ class ComputeIncoherentDOSTest(unittest.TestCase):
         ws = (energyBins[1:] + energyBins[:-1]) / 2.0
         if len(qs) > 1:
             qs = (qs[:-1] + qs[1:]) / 2.0
-        g = qs**-2 * np.exp(2 * msd * qs**2) * (1.0 - np.exp(-ws * constants.e * 1e-3 / constants.k / temperature)) * ws
+        g = qs**-2 * np.exp(msd * qs**2) * (1.0 - np.exp(-ws * constants.e * 1e-3 / constants.k / temperature)) * ws
         return g
 
     def computeFromTwoTheta(self, twoThetas, energyBins, msd=0.0, temperature=300.0):
@@ -58,7 +58,7 @@ class ComputeIncoherentDOSTest(unittest.TestCase):
         Ei = EFixed * constants.e * 1e-3
         Ef = (EFixed - ws) * constants.e * 1e-3
         qs = np.sqrt(2.0 * constants.m_n / constants.hbar**2 * ((Ei + Ef) - 2.0 * np.sqrt(Ei * Ef) * np.cos(twoTheta))) * 1e-10
-        g = qs**-2 * np.exp(2 * msd * qs**2) * (1.0 - np.exp(-ws * constants.e * 1e-3 / constants.k / temperature)) * ws
+        g = qs**-2 * np.exp(msd * qs**2) * (1.0 - np.exp(-ws * constants.e * 1e-3 / constants.k / temperature)) * ws
         return g
 
     def convertToWavenumber(self, ws):

--- a/docs/source/algorithms/ComputeIncoherentDOS-v1.rst
+++ b/docs/source/algorithms/ComputeIncoherentDOS-v1.rst
@@ -28,6 +28,12 @@ The average Debye-Waller factor :math:`\exp\left(-2\bar{W}(Q)\right)` is
 calculated using an average mean-square displacement :math:`\langle u^2 \rangle`,
 using :math:`W=Q^2\langle u^2\rangle/2`.
 
+:math:`\langle u^2 \rangle` is also called the isotropic atomic displacement
+parameter ``Uiso`` :math:`U_{\mathrm{iso}}` in Rietveld refinement programs
+like `GSAS <https://subversion.xray.aps.anl.gov/trac/EXPGUI/wiki/GSASIntro>`_.
+There is also another, related, type of atomic displacement parameter used in programs like
+`FullProf <https://www.ill.eu/sites/fullprof/>`_ called ``Biso``: :math:`B_{\mathrm{iso}}=8\pi^2U_{\mathrm{iso}}` [#GSASREF]_.
+
 The algorithm accepts both :math:`S(Q,E)` workspaces as well as
 :math:`S(2\theta,E)` workspaces. In the latter case :math:`Q` values are
 calculated from :math:`2\theta` and :math:`E` before applying the formula
@@ -109,6 +115,8 @@ References
 ----------
 
 .. [#CHAPLOT] Thermodynamic Properties of Solids, eds. Chaplot, Mittal, Choudhury, Chapter 3
+
+.. [#GSASREF] `GSAS Manual <https://subversion.xray.aps.anl.gov/EXPGUI/gsas/all/GSAS%20Manual.pdf>`_, page 123
 
 .. categories::
 

--- a/docs/source/algorithms/ComputeIncoherentDOS-v1.rst
+++ b/docs/source/algorithms/ComputeIncoherentDOS-v1.rst
@@ -29,7 +29,7 @@ calculated using an average mean-square displacement :math:`\langle u^2 \rangle`
 using :math:`W=Q^2\langle u^2\rangle/2`.
 
 :math:`\langle u^2 \rangle` is also called the isotropic atomic displacement
-parameter ``Uiso`` :math:`U_{\mathrm{iso}}` in Rietveld refinement programs
+parameter :math:`U_{\mathrm{iso}}` in Rietveld refinement programs
 like `GSAS <https://subversion.xray.aps.anl.gov/trac/EXPGUI/wiki/GSASIntro>`_.
 There is also another, related, type of atomic displacement parameter used in programs like
 `FullProf <https://www.ill.eu/sites/fullprof/>`_ called ``Biso``: :math:`B_{\mathrm{iso}}=8\pi^2U_{\mathrm{iso}}` [#GSASREF]_.

--- a/docs/source/algorithms/ComputeIncoherentDOS-v1.rst
+++ b/docs/source/algorithms/ComputeIncoherentDOS-v1.rst
@@ -32,7 +32,7 @@ using :math:`W=Q^2\langle u^2\rangle/2`.
 parameter :math:`U_{\mathrm{iso}}` in Rietveld refinement programs
 like `GSAS <https://subversion.xray.aps.anl.gov/trac/EXPGUI/wiki/GSASIntro>`_.
 There is also another, related, type of atomic displacement parameter used in programs like
-`FullProf <https://www.ill.eu/sites/fullprof/>`_ called ``Biso``: :math:`B_{\mathrm{iso}}=8\pi^2U_{\mathrm{iso}}` [#GSASREF]_.
+`FullProf <https://www.ill.eu/sites/fullprof/>`_ called :math:`B_{\mathrm{iso}}` : :math:`B_{\mathrm{iso}}=8\pi^2U_{\mathrm{iso}}` [#GSASREF]_.
 
 The algorithm accepts both :math:`S(Q,E)` workspaces as well as
 :math:`S(2\theta,E)` workspaces. In the latter case :math:`Q` values are

--- a/docs/source/release/v6.9.0/Framework/Algorithms/Bugfixes/36306.rst
+++ b/docs/source/release/v6.9.0/Framework/Algorithms/Bugfixes/36306.rst
@@ -1,0 +1,1 @@
+- Fix miscalculated Debye-Waller factor in :ref:`ComputeIncoherentDos <algm-ComputeIncoherentDos>`

